kitsune/wiki/views.py

```diff
@@ -471,106 +471,146 @@ def edit_document(request, document_slug, revision_id=None):
-                    try:
-                        doc = doc_form.save(None)
-                    except (TitleCollision, SlugCollision) as e:
-                        # TODO: .add_error() when we upgrade to Django 1.7
-                        errors = doc_form._errors.setdefault("title", ErrorList())
-                        message = "The {type} you selected is already in use."
-                        message = message.format(
-                            type="title" if isinstance(e, TitleCollision) else "slug"
-                        )
-                        errors.append(_(message))
-                    else:
-                        # Do we need to rebuild the KB?
-                        _maybe_schedule_rebuild(doc_form)
+        topics = []
+        for t in post_data.getlist("topics"):
+            topics.append(int(t))
+        post_data.setlist("topics", topics)
 
-                        return HttpResponseRedirect(
-                            urlparams(
-                                reverse("wiki.edit_document", args=[doc.slug]), opendescription=1
-                            )
-                        )
-                disclose_description = True
+        doc_form = DocumentForm(
+            post_data,
+            instance=doc,
+            can_archive=can_archive,
+            can_edit_needs_change=can_edit_needs_change,
+        )
+        if doc_form.is_valid():
+            # Get the possibly new slug for the imminent redirection:
+            try:
+                doc = doc_form.save(None)
+            except (TitleCollision, SlugCollision) as e:
+                # TODO: .add_error() when we upgrade to Django 1.7
+                errors = doc_form._errors.setdefault("title", ErrorList())
+                message = "The {type} you selected is already in use."
+                message = message.format(type="title" if isinstance(e, TitleCollision) else "slug")
+                errors.append(_(message))
             else:
-                raise PermissionDenied
-        elif which_form == "rev":
-            if doc.allows(user, "create_revision"):
-                rev_form = RevisionForm(request.POST)
-                rev_form.instance.document = doc  # for rev_form.clean()
-                if rev_form.is_valid():
-                    _save_rev_and_notify(rev_form, user, doc, base_rev=rev)
-                    if "notify-future-changes" in request.POST:
-                        EditDocumentEvent.notify(request.user, doc)
+                # Do we need to rebuild the KB?
+                _maybe_schedule_rebuild(doc_form)
 
-                    return HttpResponseRedirect(
-                        reverse("wiki.document_revisions", args=[document_slug])
-                    )
-            else:
-                raise PermissionDenied
+                return HttpResponseRedirect(
+                    urlparams(reverse("wiki.document", args=[doc.slug]), redirect="no")
```

Comment:
I'm sorry @smithellis. I'm not only doubting my suggestion to use `redirect="no"`, but realized that it doesn't even work unless we save the original slug before we save the `doc_form` on line 591, and use that original slug here instead of `doc.slug`. If you prefer to keep `redirect="no"`, you'd have to do those things, but my new suggestion is to just remove it, which means you can remove `urlparams` as well:
```suggestion
                    reverse("wiki.document", args=[doc.slug])
```

Reply:
No sweat - and I get a demerit for only running unit tests and not checking that link manually.